### PR TITLE
dcgan.ipynb Architectural and hyperparameters change

### PR DESCRIPTION
**Pull request by arshagarwal**
The following architectural and parameter changes have been made in this pull request to corroborate with the Dc-Gan paper and also to improve performance:
- LeakyRelu in the Generator has been replaced by Relu.
- Dropouts in the Discriminator have been replaced by BatchNormalization.
- Adam optimizer's beta_1 has been changed to 0.5.
- Lastly, the weights have been initialized from Normal Distribution.

**Note: All these changes significantly improve training time, for MNIST number of epochs decrease from about 25 to 15, and similar improvements can be expected with bigger datasets.**
The changes have been saved in file named **Copy_of_dcgan.ipynb** in the same folder as existing file.

### CODE REVIEW

**Comment by yashk2810** [comment]
Try to follow this: https://github.com/soumith/ganhacks

A lot of decisions in dcgan came from soumith's ganhacks who also authored dcgan.

**Comment by yashk2810** [comment]
Can you also post an image of the generated numbers image with your architecture and other datasets (as you are claiming) to verify that it is indeed better.

**Comment by lamberta** [comment]
Looks like the CI is failing.
Please format notebooks with [nbfmt](https://github.com/tensorflow/docs/blob/master/tools/tensorflow_docs/tools/nbfmt/__main__.py) and use [nblint](https://github.com/tensorflow/docs/blob/master/tools/tensorflow_docs/tools/nblint/__main__.py) to check style. Thank you

```
$ python3 -m pip install -U --user git+https://github.com/tensorflow/docs

$ python3 -m tensorflow_docs.tools.nbfmt notebook.ipynb
$ python3 -m tensorflow_docs.tools.nblint --arg=repo:tensorflow/docs notebook.ipynb
```

**Comment by arshagarwal** [comment]
> Try to follow this: https://github.com/soumith/ganhacks
> 
> A lot of decisions in dcgan came from soumith's ganhacks who also authored dcgan.

Thanks for sharing the link, I agree that the hacks presented in the link are very insightful and backed by evidence, however I feel that the tutorial is aimed at introducing people to dcgan, and thus should abide by dcgan constraints verbatim, anything else can become a little misleading for people new to dcgan.

**Comment by arshagarwal** [comment]
> Can you also post an image of the generated numbers image with your architecture and other datasets (as you are claiming) to verify that it is indeed better.

Here are the [dcgan results](https://github.com/arshagarwal/docs/tree/arsh_test/site/en/tutorials/generative/dcgan%20results) as requested.
The current and the new results are stored in **Images_curr** and **Images_new** directories respectively.

**Comment by arshagarwal** [comment]
> Looks like the CI is failing.
> Please format notebooks with [nbfmt](https://github.com/tensorflow/docs/blob/master/tools/tensorflow_docs/tools/nbfmt/__main__.py) and use [nblint](https://github.com/tensorflow/docs/blob/master/tools/tensorflow_docs/tools/nblint/__main__.py) to check style. Thank you
> 
> ```
> $ python3 -m pip install -U --user git+https://github.com/tensorflow/docs
> 
> $ python3 -m tensorflow_docs.tools.nbfmt notebook.ipynb
> $ python3 -m tensorflow_docs.tools.nblint --arg=repo:tensorflow/docs notebook.ipynb
> ```

Thanks for sharing the CI tools, I have formatted the notebook to nbft format, however I am afraid I am  unable to fix the nbilnt style.
Running the aforementioned command throws the following error.
```
Lint notebook: Copy_of_dcgan.ipynb
Pass | tensorflow::not_translation
Fail | google::second_person | Prefer second person 'you' instead of 'we': https://developers.google.com/style/person
Pass | tensorflow::license_check | Apache license is required
Pass | tensorflow::button_r1_extra | Remove extra buttons from TF 1.x docs.
Fail | tensorflow::button_colab | Missing or malformed URL in Colab button.
Fail | tensorflow::button_download | Missing or malformed URL in Download button.
Fail | tensorflow::button_github | Missing or malformed URL in GitHub button.
Fail | tensorflow::button_website | Missing or malformed URL in 'View on' button.
Pass | tensorflow::copyright_check | Copyright required

Lint log:
[tensorflow::button_download] Download button URL doesn't match: https://storage.googleapis.com/tensorflow_docs/docs/site/en/tutorials/generative/Copy_of_dcgan.ipynb
[tensorflow::button_website] 'View on' button URL doesn't match pattern: https://www.tensorflow.org/.*tutorials/generative/Copy_of_dcgan
[tensorflow::button_colab] Colab button URL doesn't match: https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/generative/Copy_of_dcgan.ipynb
[tensorflow::button_github] GitHub button URL doesn't match: https://github.com/tensorflow/docs/blob/master/site/en/tutorials/generative/Copy_of_dcgan.ipynb
```
Could you share how I could fix these.

**Comment by yashk2810** [comment]
According to that ganhacks link, dcgan doesn't work properly without the hacks with other datasets. So I don't want to introduce anyone to the verbatim paper. We link off at the top to the paper and then modify the architecture slightly to give stable training. So it would be nice to stick to the hacks.

**Comment by arshagarwal** [comment]
> According to that ganhacks link, dcgan doesn't work properly without the hacks with other datasets. So I don't want to introduce anyone to the verbatim paper. We link off at the top to the paper and then modify the architecture slightly to give stable training. So it would be nice to stick to the hacks.

I agree with you introducing people to newer research is even better, but all I am requesting is a simple heads up.
And I went through the link a couple times, but I am afraid I am unable to find evidence that claims that dcgan doesn't work properly without the hacks, besides I have also shared  better results without following those hacks for MNIST dataset.

Could you share where exactly in the link it says that Gan doesn't work properly without the proposed hacks?

**Comment by arshagarwal** [comment]
Also I reckon initializing the weights from normal distribution and not glorot uniform which is used by default for initialization should be introduced in the tutorial, because that has not been negated yet, to the best of my knowledge.

**Comment by arshagarwal** [comment]
> Please don't commit files not intended for review: remove images and copies not meant for publication.

I have deleted redundant files, sorry for the inconvenience.

**Comment by MarkDaoust** [comment]
@yashk2810, could you make a decision on this? Please get it closed or merged.

File: site/en/tutorials/generative/dcgan.ipynb [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+  },
   "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "view-in-github",
+        "colab_type": "text"
+      },
+      "source": [
+        "<a href=\"https://colab.research.google.com/github/arshagarwal/docs/blob/arsh_test/site/en/tutorials/generative/dcgan.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
Comment (yashk2810) [inline site/en/tutorials/generative/dcgan.ipynb:26]: Please revert this.

File: site/en/tutorials/generative/dcgan.ipynb [hunk 2]
```diff
@@ -93,11 +119,11 @@
         "## What are GANs?\n",
         "[Generative Adversarial Networks](https://arxiv.org/abs/1406.2661) (GANs) are one of the most interesting ideas in computer science today. Two models are trained simultaneously by an adversarial process. A *generator* (\"the artist\") learns to create images that look real, while a *discriminator* (\"the art critic\") learns to tell real images apart from fakes.\n",
         "\n",
-        "![A diagram of a generator and discriminator](./images/gan1.png)\n",
+        "![A diagram of a generator and discriminator](https://github.com/tensorflow/docs/blob/master/site/en/tutorials/generative/images/gan1.png?raw=1)\n",
```
Comment (yashk2810) [inline site/en/tutorials/generative/dcgan.ipynb:122]: Revert this

**Comment by yashk2810** [comment]
There are lot of unncessary changes in this PR. Please revert them and run your notebook via `nbfmt`. You can check the Github Actions failure to see what failed and what to run.

**Comment by yashk2810** [comment]
> Adam optimizer's beta_1 has been changed to 0.5.
> Lastly, the weights have been initialized from Normal Distribution.

Let's go with these two changes. You can revert the other changes.

**Comment by arshagarwal** [comment]
I have made the changes you asked in this [pull request](https://github.com/tensorflow/docs/pull/1715).
I am sure there is a more graceful way to do this, but I found this way much easier than to fix code in the same pull request.
Also, I hope this will make it easier for you to compare the two branches.
Sorry about that!

**Comment by MarkDaoust** [comment]
Thanks. Let's move the conversation over there.
